### ISSUE TITLE
Error when canvas file is in the folder

### ISSUE THREAD

**Issue by anareaty** [lynchjames/obsidian-writing-goals#50]
I add word count goal to the folder. It works correctly. But if I add the canvas file to this folder and reload Obsidian, word count stops showing and this error shows up:

```
plugin:writing-goals:20432 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'contains')
    at GoalHelper.additionalFileTypes (plugin:writing-goals:20432:50)
    at GoalHelper.isValidFile (plugin:writing-goals:20429:98)
    at GoalHelper.getWordCountRecursive (plugin:writing-goals:20416:18)
    at GoalHelper.getWordCountRecursive (plugin:writing-goals:20423:44)
    at GoalHelper.createGoal (plugin:writing-goals:20340:87)
    at GoalHelper.updateGoalsFromSettings (plugin:writing-goals:20408:37)
    at async WritingGoals.loadNoteGoalData (plugin:writing-goals:22339:9)
```
I didn't add anything to the additional file types setting, so I expected canvas files to be ignored. Instead they are breaking something.

**Comment by pexfapro** [comment]
Also seeing this issue. Any resolution @anareaty ?

**Comment by Azmoinal** [comment]
Could you try to explicitly put md into the field “Additional file types” under Writing goals settings?
I faced the same issue as here for canvas using the “Edit History” plugin that creates .edtz files in the same folder as updates file to maintain a local history. 
Enabling this plugin, writing goals stopped up I manually remove any edtz file. 

As a tentative, I explicitly put md as only file to check (without comma) and now canvas file and edtz seem to no longer interfere with Writing Goals. 

Let me know if it works for you.

**Comment by Medullitus** [comment]
Is the developer looking here?? I'm getting a 'FAILED TO SHOW GOAL' error!!

**Comment by Medullitus** [comment]
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'dailyGoalCount')
    at create_fragment$2 (plugin:writing-goals:4151:33)
    at init (plugin:writing-goals:1018:34)
    at new Simple_goal (plugin:writing-goals:4359:3)
    at FileLabels.createSimpleGoal (plugin:writing-goals:4392:9)
    at eval (plugin:writing-goals:4386:26)
    at Array.forEach (<anonymous>)
    at FileLabels.initFileLabels (plugin:writing-goals:4378:21)
    at WritingGoals.loadNoteGoalData (plugin:writing-goals:22341:29)
    at async GoalModal.onSubmit (plugin:writing-goals:20584:9)
create_fragment$2 @ plugin:writing-goals:4151
init @ plugin:writing-goals:1018
Simple_goal @ plugin:writing-goals:4359
createSimpleGoal @ plugin:writing-goals:4392
eval @ plugin:writing-goals:4386
initFileLabels @ plugin:writing-goals:4378
loadNoteGoalData @ plugin:writing-goals:22341

**Comment by Medullitus** [comment]
UPDATE: I found the issue! The HOMEPAGE plugin

When the Homepage plugin is enabled, the daily word goal plugin doesn't open and gives an error. You can disable the Homepage plugin, open the word goal window, and then reactivate the Homepage plugin. Thanks. @lynchjames  we love you and your plugin is amazing! Please don't leave us!! Thanks <3<3<3<3<3

**Comment by AustinBrrtt** [comment]
I had this issue and adding just `md` does allow the plugin to work on folders with canvases, ignoring the canvas content.

I also found that putting `md,canvas` allowed it to work and count the canvas, but the word count for the canvas counts words in the JSON file for the canvas rather than parsing it to get the word count for just the content, which at least when I tested it gives 13 extra words per node.

It wouldn't be too hard to add support for counting words in canvas files by just parsing the JSON and adding up the word count for the `text` property of any object in the `nodes` array that has one. The Novel Word Count plugin added [this behavior](https://github.com/isaaclyman/novel-word-count-obsidian/blob/master/logic/canvas.ts) back in February for example.
